tests/python_client/testcases/test_query.py

```diff
@@ -2854,3 +2792,125 @@ def test_count_expression_comparative(self):
+        collection_w.load()
+        # 2. search iterator
+        expr = "int64 >= 0"
+        collection_w.query_iterator(expr, limit=limit, offset=offset,
+                                    check_task=CheckTasks.check_query_iterator,
+                                    check_items={"count": ct.default_nb - offset,
+                                                 "limit": limit})
+
+    @pytest.mark.tags(CaseLabel.L0)
+    @pytest.mark.parametrize("limit", [100, 500])
+    @pytest.mark.parametrize("auto_id", [True, False])
+    def test_query_iterator_empty_expr(self, auto_id, limit):
+        """
+        target: test query iterator with empty expression
+        method: query iterator empty expression with a limit
+        expected: return topK results by order
+        """
+        # 1. initialize with data
+        collection_w, _, _, insert_ids = self.init_collection_general(prefix, True, auto_id=auto_id)[0:4]
+
+        # 2. query with limit
+        collection_w.query_iterator("", limit=limit,
+                                    check_task=CheckTasks.check_query_iterator,
+                                    check_items={"limit": limit,
+                                                 "count": ct.default_nb,
+                                                 "exp_ids": insert_ids})
+
+    @pytest.mark.tags(CaseLabel.L2)
+    @pytest.mark.parametrize("offset", [100, 1000])
+    @pytest.mark.parametrize("limit", [500, 1000])
+    def test_query_iterator_expr_empty_with_random_pk_pagination(self, limit, offset):
+        """
+        target: test query iterator with empty expression
+        method: create a collection using random pk, query empty expression with a limit
+        expected: return topK results by order
+        """
+        # 1. initialize with data
+        collection_w = self.init_collection_general(prefix, with_json=False)[0]
+
+        # 2. generate unordered pk array and insert
+        unordered_ids = [i for i in range(ct.default_nb)]
+        random.shuffle(unordered_ids)
+        float_value = [np.float32(i) for i in unordered_ids]
+        string_value = [str(i) for i in unordered_ids]
+        vector_value = cf.gen_vectors(nb=ct.default_nb, dim=ct.default_dim)
+        collection_w.insert([unordered_ids, float_value, string_value, vector_value])
+        collection_w.load()
+
+        # 3. query with empty expr and check the result
+        exp_ids = sorted(unordered_ids)
+        collection_w.query_iterator("", limit=limit, output_fields=[ct.default_string_field_name],
+                                    check_task=CheckTasks.check_query_iterator,
+                                    check_items={"limit": limit, "count": ct.default_nb, "exp_ids": exp_ids})
+
+        # 4. query with pagination
+        exp_ids = sorted(unordered_ids)[offset:]
+        collection_w.query_iterator("", limit=limit, offset=offset, output_fields=[ct.default_string_field_name],
```

Comment:
do we have case covering out of max limit or offset? If not, could add a case.

Reply:
Added.